core/input/input_event.cpp

```diff
@@ -1494,3 +1495,27 @@ void InputEventMIDI::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "controller_number"), "set_controller_number", "get_controller_number");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "controller_value"), "set_controller_value", "get_controller_value");
 }
+
+///////////////////////////////////
+
+void InputEventShortcut::set_shortcut(Ref<Shortcut> p_shortcut) {
+	ERR_FAIL_COND(Object::cast_to<InputEventShortcut>(*p_shortcut) != nullptr);
```

Comment:
I don't get this condition. `Shortcut` will never be `InputEventShortcut`, so not sure why are you checking this. Unless there is some implicit conversion I'm not aware of 🤔

Reply:
so that check was just to make sure that `InputEventShortcut::shortcut` value to never get assigned with an `InputEventShortcut`. don't know if that is really possible but as it is a public function checking it is good right?

Reply:
But these two are completely different types, you'd get compilation error on wrong use. The check will never be true. (You could try for yourself)

Reply:
@KoBeWi is right here. This check is useless.
I believe you wrongly interpreted my advice. This check needs to be inside the `ShortCut::set_shortcut` function, as it's place where we do not want to assign such event to a shortcut.

Also, `ERR_FAIL_COND(Object::cast_to<InputEventShortcut>(*p_shortcut))` is enough, no need to add`!= nullptr`. (casting nullptr to a bool result in `false`)

Reply:
Ok